target/linux/mvebu/image/wdmc.mk

```diff
@@ -0,0 +1,21 @@
+ifeq ($(SUBTARGET),wdmc)
+
+define Device/armada-385-wd-mcex2u
+  UBINIZE_OPTS := -E 5
+  BLOCKSIZE := 128k
+  PAGESIZE := 2048
+
+  IMAGES += image.cfs-factory.img
+  IMAGE/image.cfs-factory.img := append-ubi
+  IMAGES += uImage-factory.img
+  IMAGE/uImage-factory.img := append-kernel
+  IMAGES += sysupgrade.bin
+  IMAGE/sysupgrade.bin := append-kernel | sysupgrade-tar kernel=$$$$@ | append-metadata
+
+  KERNEL_SIZE := 5120k
+  DEVICE_TITLE := WD My Cloud EX2 Ultra (DB-88F6820)
+  DEVICE_PACKAGES += wtmcu -dnsmasq -ppp
```

Comment:
BTW where is this wtmcu package? I can't find it in this PR or in the OpenWrt main/packages trees.

Reply:
Here: openwrt/packages#8929. I'm moving it into this PR, as was suggested.

Reply:
Well I didn't suggested this, I was just wondering where it is as I've never heard about it before and couldn't find it either. Anyway, as you can't depend in the image on the package which is not part of OpenWrt main tree, it should be included as part of this PR, so whoever suggested it, was mostly correct.

By mostly above I mean, that I've just quickly looked at source code of that wtmcu package and it seems to me, that most of this functionality belongs to the kernel instead to the userspace application. It should be probably implemented in the same spirit as your current `wdmcex2u,power-off` so you would need implement driver for `wdmcex2u,leds`, `wdmcex2u,fan` and `wdmcex2u,hwmon` (just my very rough estimation and upstream might see it differently).

You can ask upstream developers for suggestions once you start upstreaming that `wdmcex2u,power-off` driver and/or the DTS for this device.

Reply:
I've considered this approach. And while fan and hwmon related functions are feasible as a driver, things like WOL (is not enough to just prime phy device, MCU has another switch), wake-up alarm and LEDs (fixed modes and patterns, no way to control brightness, tty exchange speed won't allow them to really 'blink' in a custom way) are not so fitted into driver models as far I can imagine. 

Also, I could not find any intention from WD or Weltrend to make a proper driver for this MCU or at least open it's specification. I'm afraid I won't have enough resources to implement it all by myself.

The wdmcex2u,power-off is really a simple hack and pretty much essential to be a driver for the board to be adequately functional (even then WD uses userspace tool for that, lot's of WOL bugs related to improper shutdown, their firmware is an abomination). I'm not sure if upstream will accept a bunch hacks like that applied to variety of drivers like net/phy/marvell as they are very board-specific.